test/test_dropbox.py

```diff
@@ -137,5 +144,57 @@ def test_team(self, dbxt):
             team_member_id = r.members[0].profile.team_member_id
             dbxt.as_user(team_member_id).files_list_folder('')
 
+    @dbx_from_env
+    def test_with_path_root_constructor(self, dbx):
+        # Verify valid mode types
+        for mode, ns_id in (
+                ("home", None),
+                ("root", "123"),
+                ("namespace_id", "123")
+        ):
+            dbx_new = dbx.with_path_root(mode, ns_id)
+            self.assertIsNot(dbx_new, dbx)
+
+            path_root_mode = getattr(PathRoot, mode)
+            path_root = path_root_mode(ns_id) if ns_id else path_root_mode
+
+            expected = stone_serializers.json_encode(PathRoot_validator, path_root)
+            self.assertEqual(dbx_new._headers.get(PATH_ROOT_HEADER), expected)
+
+        # verify invalid mode raises ValueError
+        with self.assertRaises(ValueError):
+            dbx.with_path_root("invalid_mode")
+
+    @dbx_from_env
+    def test_path_root(self, dbx):
+        root_info = dbx.users_get_current_account().root_info
+        root_ns = root_info.root_namespace_id
+        home_ns = root_info.home_namespace_id
+
+        # verify home mode
+        dbxpr = dbx.with_path_root('home')
+        dbxpr.files_list_folder('')
+
+        # verify root mode
+        dbxpr = dbx.with_path_root("root", root_ns)
+        dbxpr.files_list_folder('')
```

Comment:
Does this work with non-CDM users? I _think_ it will but I don't know for sure. (I've kind of lost track of the evolution of this stuff.)

Reply:
Yup, works for both CDM and non.